HIP/hip-1261.md

```diff
@@ -0,0 +1,847 @@
+| Allowances                 | Number of allowances.                                                                                                                                                                                                             |
+| Airdrops                   | Number of airdrops executed.                                                                                                                                                                                                      |
+
+### Node
+
+The *node* fee is computed exactly the same way for all transactions. It is therefore defined in one place in the
+configuration and applicable to all transactions.
+
+```json
+{
+  "node": {
+    "base": 100000,
+    "extras": [
+      {"name": "Bytes", "includedCount": 1024},
+      {"name": "Signatures", "includedCount": 1}
+    ]
+  }
+}
+```
+
+### Network
+
+The *network* fee is also computed exactly the same way for all transactions. It is also defined in one place in the
+configuration and applicable to all transactions. It is specified as a multiplier of the node fee.
+
+```json
+{
+  "network": {"multiplier": 9}
+}
+```
+
+### Services
+
+A service is a grouping of transactions and queries. These match the gRPC **services defined in the HAPI specification.
+Each transaction or query is given a `name`, `base`, and a list of the `extras` that apply to that transaction. For each
+such `extra`, an `includedCount` *may* be specified, indicating how many of that extra are included in the `base`.
+
+```json
+{
+  "name": "CryptoService",
+  "transactions": [
+    {
+      "name": "CryptoCreate",
+      "base": 499000000,
+      "extras": [
+        {
+          "name": "Keys",
+          "includedCount": 1
+        }
+      ]
+    }
+  ],
+  "queries": []
+}
+```
+
+<aside>
+💡
+
+NOTE: In this transaction we see that the base price is `499000000` tinycents, or `0.0499` USD. At runtime, for a basic
+`CryptoCreate` transaction, the total fee will be *node* + *network* + *service:*
+
+```text
+totalChargedKeys = (max(1, numKeys) - 1) * costPerKey;
+```
+
+Since `Keys` has an `includedCount` of 1, if the transaction only sets a single key on the new account, then it is not
+charged for that key. But if it were to set two keys, then it would pay for one. In other words:
+
+```text
+totalFee = 100000 + 900000 + 499000000
+```
+
+</aside>
+
+`extras` are referred to by name. The `includedCount`, if specified, must be a non-negative integer value. If not
+specified, the default value for an `includedCount` is 0.
+
+### Unreadable Transactions
+
+```json
+{
+  "unreadable": {"fee": 100000000000}
+}
+```
+
+Since an unreadable transaction could not even be parsed, it is not possible for it to have any extras. It therefore has
+a single fee, charged to the node that sent this unreadable transaction. It is intended to be punitive, but is up to the
+network owners to determine the appropriate price.
+
+### Queries
+
```

Comment:
The updated protobuf does not distinguish between services and queries. Queries can use  ServiceFeeDefinition.free = true to indicate they are free. Do we still need to keep the separate?

Reply:
Per the meeting today, we will keep the service transactions and and queries as the same type and not break them out separately in the protobuf. The prose should be updated to reflect this.